File: frontend/lib/models/category.dart
class CategoryLevel {
  final int catLevelId;
  final String catLevelName;
  final int catLevelTtlMins;

  CategoryLevel({
    required this.catLevelId,
    required this.catLevelName,
    required this.catLevelTtlMins,
  });

  factory CategoryLevel.fromJson(Map<String, dynamic> json) {
    return CategoryLevel(
      catLevelId: json['cat_level_id'] as int,
      catLevelName: json['cat_level_name'] as String,
      catLevelTtlMins: json['cat_level_ttl_mins'] as int,
    );
  }
}

class Category {
  final int catId;
  final int catLevelId;
  final String catName;

  Category({
    required this.catId,
    required this.catLevelId,
    required this.catName,
  });

  factory Category.fromJson(Map<String, dynamic> json) {
    return Category(
      catId: json['cat_id'] as int,
      catLevelId: json['cat_level_id'] as int,
      catName: json['cat_name'] as String,
    );
  }
}

class SubCategory {
  final int subCatId;
  final int catId;
  final String subCatName;

  SubCategory({
    required this.subCatId,
    required this.catId,
    required this.subCatName,
  });

  factory SubCategory.fromJson(Map<String, dynamic> json) {
    return SubCategory(
      subCatId: json['sub_cat_id'] as int,
      catId: json['cat_id'] as int,
      subCatName: json['sub_cat_name'] as String,
    );
  }
}

File: backend/schemas/Category.py
from pydantic import BaseModel, ConfigDict


class CategoryLevelResponse(BaseModel):
    """Schema for category level response"""
    cat_level_id: int
    cat_level_name: str
    cat_level_ttl_mins: int

    model_config = ConfigDict(from_attributes=True)


class CategoryResponse(BaseModel):
    """Schema for category response"""
    cat_id: int
    cat_level_id: int
    cat_name: str

    model_config = ConfigDict(from_attributes=True)


class SubCategoryResponse(BaseModel):
    """Schema for sub category response"""
    sub_cat_id: int
    cat_id: int
    sub_cat_name: str

    model_config = ConfigDict(from_attributes=True)
